manifests/cluster/backup/mysqldump.pp

```diff
@@ -0,0 +1,120 @@
+# mariadb::cluster::backup::mysqldump
+#
+# Manage MariaDB Galera Cluster Backup
+#
+# @summary Manage MariaDB Galera Cluster Backup
+#
+# @example
+#   include mariadb::cluster::backup::mysqldump
+class mariadb::cluster::backup::mysqldump (
+  $backupuser         = '', # lint:ignore:empty_string_assignment
+  $backuppassword     = '', # lint:ignore:empty_string_assignment
+  $backupdir          = '', # lint:ignore:empty_string_assignment
+  $maxallowedpacket   = '1M',
+  $backupdirmode      = '0700',
+  $backupdirowner     = $mariadb::params::user,
+  $backupdirgroup     = $mariadb::params::group,
+  $backupcompress     = true,
+  $backuprotate       = 30,
+  $initiator_node     = false,
+  $ignore_events      = true,
+  $galara_port        = '4567',
+  $delete_before_dump = false,
+  $backupdatabases    = [],
+  $include_triggers   = false,
+  $include_routines   = false,
+  $ensure             = 'present',
+  $time               = ['23', '5'],
+  $prescript          = false,
+  $postscript         = false,
+  $execpath           = '/usr/bin:/usr/sbin:/bin:/sbin',
+  $optional_args      = [],
+) inherits mariadb::params {
+
+  include '::mariadb::cluster'
+
+  if $backupcompress {
+    ensure_packages(['bzip2'])
+    Package['bzip2'] -> File['wsrep_sst_backup']
+  }
+
+  $initiator_ensure = $initiator_node ? {
+    true    => $ensure,
+    default => 'absent',
+  }
+
+  $galera_options = $mariadb::cluster::galera_config::options['mysqld']
+
+  mysql_user { "${backupuser}@localhost":
+    ensure        => $ensure,
+    password_hash => mysql_password($backuppassword),
+    require       => Class['mysql::server::root_password'],
+  }
+
+  if $include_triggers  {
+    $privs = [ 'SELECT', 'RELOAD', 'LOCK TABLES', 'SHOW VIEW', 'PROCESS', 'TRIGGER' ]
+  } else {
+    $privs = [ 'SELECT', 'RELOAD', 'LOCK TABLES', 'SHOW VIEW', 'PROCESS' ]
+  }
+
+  mysql_grant { "${backupuser}@localhost/*.*":
+    ensure     => $ensure,
+    user       => "${backupuser}@localhost",
+    table      => '*.*',
+    privileges => $privs,
+    require    => Mysql_user["${backupuser}@localhost"],
+  }
+
+  $garb_address = join($mariadb::cluster::wsrep_cluster_peers.suffix(":${galara_port}"), ',')
```

Comment:
We can't use cluster_peers, b/c its not guaranteed to be set.

Reply:
you needed the port on each address...

Reply:
Lets go back to using the address from the galera_config, but make it add the port also.

Reply:
So add your suffix here: https://github.com/kobybr/puppet-mariadb/blob/e98d2a6789ae70c12eabb9d3969ba837f8c02936/manifests/cluster/galera_config.pp#L16